src/metaswitch/clearwater/config_manager/config_access.py

```diff
@@ -343,19 +353,48 @@ def main(args):
         try:
             upload_verified_config(config_loader,
                                    local_store,
-                                   args.config_type,
+                                   selected_config,
                                    args.force,
                                    args.autoconfirm)
         except (UserAbort, ConfigUploadFailed) as exc:
             log.error("Upload failed")
             sys.exit(exc)
 
 
+def lookup_config_type(config_type, config_location):
+    """
+    This takes the string defining the config_type from arg.parse and returns
+    a instance of the class containing data on the config type selected
```

Comment:
A few issues to fix here:
- It's still unclear what `config_location` represents here. Why is it passed in?
- It feels a bit odd that you require that the config_type string comes from `arg.parse`. (a) I don't believe `arg.parse` is ever actually called anywhere in the code, and (b) ideally, your functions shouldn't care where their inputs come from.
- `a instance` -> `an instance`